recipe/build.sh

```diff
@@ -6,16 +6,24 @@ set +ex
 # https://gitlab.kitware.com/cmake/cmake/blob/master/Modules/FindPNG.cmake#L55
 ln -s $PREFIX/include/libpng16 $PREFIX/include/libpng
 
+# Be safe, search and replace all the occurances I found
+sed -i.bak "s/CXX_STANDARD 11/CXX_STANDARD 17/" platforms/android/build-tests/test_cmake_build.py
+sed -i.bak "s/CXX_STANDARD 11/CXX_STANDARD 17/" modules/dnn/cmake/plugin.cmake
+sed -i.bak "s/CXX_STANDARD 11/CXX_STANDARD 17/" modules/highgui/cmake/plugin.cmake
+sed -i.bak "s/CXX_STANDARD 11/CXX_STANDARD 17/" modules/gapi/cmake/standalone.cmake
+sed -i.bak "s/CXX_STANDARD 11/CXX_STANDARD 17/" modules/videoio/cmake/plugin.cmake
+sed -i.bak "s/CXX_STANDARD 11/CXX_STANDARD 17/" cmake/OpenCVDetectCXXCompiler.cmake
+sed -i.bak "s/CXX_STANDARD 11/CXX_STANDARD 17/" cmake/OpenCVPluginStandalone.cmake
+sed -i.bak "s/CXX_STANDARD 11/CXX_STANDARD 17/" modules/objc/test/cmakelists.template
+sed -i.bak "s/CXX_STANDARD 11/CXX_STANDARD 17/" modules/objc/generator/templates/cmakelists.template
```

Comment:
is this stuff necessary still?

Reply:
I removed and almost all jobs are passing, so I guess no.

Reply:
yeah, i'm somewhat worried it is necessary for osx lol ;) you never know!

Reply:
i'm somewhat worried it is due to the order of arguments. 

I think protobuf now expeoses abseil (right???) and thus needs a consistent version of c++ between libraries.

this is why I originally had added it (before your patch)

Reply:
it may be that for all other platforms except for osx arm64, the order of the c++ flags makes it so that it is c++17 and not c++11

Reply:
The `osx-arm64` failure is due to the use of the host protoc instead of the build one:
~~~
2023-06-05T08:59:56.3610410Z [552/1488] cd $SRC_DIR/build/modules/dnn && $PREFIX/bin/protoc-23.2.0 --cpp_out :$SRC_DIR/build/modules/dnn -I $SRC_DIR/modules/dnn/src/caffe -I $SRC_DIR/modules/dnn/src/onnx -I $SRC_DIR/modules/dnn/src/tensorflow $SRC_DIR/modules/dnn/src/caffe/opencv-caffe.proto
2023-06-05T08:59:56.3612450Z FAILED: modules/dnn/opencv-caffe.pb.h modules/dnn/opencv-caffe.pb.cc $SRC_DIR/build/modules/dnn/opencv-caffe.pb.h $SRC_DIR/build/modules/dnn/opencv-caffe.pb.cc 
2023-06-05T08:59:56.3614230Z cd $SRC_DIR/build/modules/dnn && $PREFIX/bin/protoc-23.2.0 --cpp_out :$SRC_DIR/build/modules/dnn -I $SRC_DIR/modules/dnn/src/caffe -I $SRC_DIR/modules/dnn/src/onnx -I $SRC_DIR/modules/dnn/src/tensorflow $SRC_DIR/modules/dnn/src/caffe/opencv-caffe.proto
2023-06-05T08:59:56.3615580Z /bin/sh: $PREFIX/bin/protoc-23.2.0: Bad CPU type in executable
~~~

see https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/4075#issuecomment-1569242816 for a possible patch. I do not think there is any problem of c++ flags order, if you look to the compiler invocation each invocation either has `-std=c++11` or `-std=c++17`. As protobuf or abseil are not exposed in public headers, there is no risk of this creating problems. However, if we want to just play safe, we can re-add the patch to ensure that all of opencv compilers with C++17.

Reply:
The osx-arm64 failures hopefully will be fixed by https://github.com/conda-forge/opencv-feedstock/pull/363/commits/2436f99f54347ec646d0e3d7dc8c9be524aba7a3 .

Reply:
> I do not think there is any problem of c++ flags order, if you look to the compiler invocation each invocation either has `-std=c++11` or `-std=c++17`

nope nope nope nope, please let's not play this game of whack-a-mole with abseil (which only supports a consistent C++ standard). 🙏 

> As protobuf or abseil are not exposed in public headers

They've mentioned they plan to use abseil in their public API, so this will happen eventually.

Reply:
> They've mentioned they plan to use abseil in their public API, so this will happen eventually.

Ah sorry, I got mixed up here. Protobuf will expose abseil in its API, opencv may choose not to do that. Still, I'd strongly advise that we use a uniform C++ standard version here.

Reply:
> > if you look to the compiler invocation each invocation either has `-std=c++11` or `-std=c++17`
> 
> nope nope nope nope, please let's not play this game of whack-a-mole with abseil (which only supports a consistent C++ standard). 🙏

Sure, I was just replying to @hmaarrfk on the point of the order of the arguments, not endorsing mixing standard versions in the same library (that unfortunatly it is what happens if one set manually `CMAKE_CXX_STANDARD` and/or `CXX_STANDARD` .